packages/gitbook/src/routes/markdownPage.ts

```diff
@@ -0,0 +1,34 @@
+import { resolvePagePath } from '@/lib/pages';
+import { RevisionPageType } from '@gitbook/api';
+import type { GitBookSiteContext } from '@v2/lib/context';
+import { throwIfDataError } from '@v2/lib/data';
+
+/**
+ * Generate a markdown version of a page.
+ */
+export async function servePageMarkdown(context: GitBookSiteContext, pagePath: string) {
+    const pageLookup = resolvePagePath(context.pages, pagePath);
+    if (!pageLookup) {
+        return new Response('Page not found', { status: 404 });
+    }
+
+    const { page } = pageLookup;
+
+    if (page.type !== RevisionPageType.Document) {
+        return new Response('Page is not a document', { status: 404 });
```

Comment:
```suggestion
        return new Response('Page is not a document', { status: 400 });
```

Just so that tools that rely on the 404 don't get confused thinking the page doesn't exist?

Reply:
Actually you could argue the Markdown doesn't exist so 404 is probably fine